### PR TITLE
change unicode filter method

### PR DESCRIPTION
**Pull request by WMDE-Fisch**
addresses comments in https://github.com/wmde/DeepCat-Gadget/pull/65
( \u200B ) will _not_ be filtered

replaced whitespace filter according to https://github.com/wikimedia/mediawiki/blob/master/includes/title/MediaWikiTitleCodec.php#L227

### CODE REVIEW

**Comment by wmde-manicki** [comment]
apart from minor comments +1

**Comment by WMDE-Fisch** [comment]
/me waiting for @thiemowmde s comments ;-)

File: DeepCat.js [hunk 1]
```diff
@@ -584,16 +584,27 @@
...
 	}
 
 	/**
+	 * Replace whits-paces with underscores
+	 * @see https://github.com/wikimedia/mediawiki/blob/master/includes/title/MediaWikiTitleCodec.php#L227
+	 * @param {string} str
+	 * @return {string}
+	 */
+	function replaceWhiteSpace( str ) {
+		var re = /[ _\u00A0\u1680}\u180E\u2000-\u200A\u2028\u2029\u202F\u205F\u3000]+/g;
```
Comment (thiemowmde) [inline DeepCat.js]: I believe the `}` is a mistake.

**Comment by thiemowmde** [comment]
Seems this now replicates what core does. Like that. :+1:

File: DeepCat.js (hunk 1 above)
Comment (WMDE-Fisch) [inline DeepCat.js]: d'oh thx, fixed

**Pull request merged**